src/sankoff_handler.cpp

```diff
@@ -3,103 +3,158 @@
 
 #include "sankoff_handler.hpp"
 
-Eigen::Matrix<double, Eigen::Dynamic, SankoffHandler::state_count> SankoffHandler::GenerateLeavesPartials(const size_t node_count, const size_t pattern_idx) {
-    auto site_partials = Eigen::Matrix<double, Eigen::Dynamic, state_count>();
-    site_partials.resize(node_count, state_count);
+SankoffPartialVec SankoffHandler::GenerateLeafPartials(const size_t node_count) {
+  SankoffPartialVec p_partials_left(site_pattern_.PatternCount());
+  for (size_t pattern_idx = 0; pattern_idx < site_pattern_.PatternCount();
+       pattern_idx++) {
+    // p_partials_left[pattern_idx] =
+    //     GenerateLeafPartials(topology->Id() + 1, pattern_idx);
+    SankoffPartial site_partials(node_count, state_count);
     site_partials.setZero();
     for (size_t seq = 0; seq < site_pattern_.TaxonCount(); seq++) {
-        auto site_val = site_pattern_.GetPatternSymbol(seq, pattern_idx);
-        if (site_val < state_count) {
-            for (size_t i = 0; i < state_count; i++) {
-                if (i != site_val) {
-                    site_partials(seq, i) = static_cast<double>(INT_MAX);
-                }
-            }
+      auto site_val = site_pattern_.GetPatternSymbol(seq, pattern_idx);
+      if (site_val < state_count) {
+        for (size_t i = 0; i < state_count; i++) {
```

Comment:
This is so not important, but one could set everything to a constant `big_double` value and then come through and just change the `site_val` entry to 0.